copy.lua

```diff
@@ -0,0 +1,417 @@
+--designating tool
+
+--[====[
+
+copy
+====
+copy is a gui program designed to make designating easier and more efficient.
+
+]====]
+
+--[[
+ Author's Note:
+ 
+ shoutout to twitch.tv/keupo
+ created by flavorstreet
+ 
+ pipicaca.
+]]
+local utils = require 'utils'
+local gui = require 'gui'
+local guidm = require 'gui.dwarfmode'
+local dlg = require 'gui.dialogs'
+
+CopyUI = defclass(CopyUI, guidm.MenuOverlay)
+
+CopyUI.ATTRS {
+    state='none',
+    buffer=nil,
+    offsetDirection=0,
+    cull=true,
+    blink=false,
+    option='normal'
+}
+local digSymbols={' ', 'X', '_', 30, '>', '<'}
+function CopyUI:init()
+    self.saved_mode = df.global.ui.main.mode
+end
+
+function CopyUI:onShow()
+    df.global.ui.main.mode = df.ui_sidebar_mode.Stockpiles
+end
+
+function CopyUI:onDestroy()
+    df.global.ui.main.mode = self.saved_mode
+end
+
+local function getTileType(cursor)
+    local block = dfhack.maps.getTileBlock(cursor)
+    if block then
+        return block.tiletype[cursor.x%16][cursor.y%16]
+    else
+        return 0
+    end
+end
+
+local function paintMapTile(dc, vp, cursor, pos, ...)
+    if not same_xyz(cursor, pos) then
+        local stile = vp:tileToScreen(pos)
+        if stile.z == 0 then
+            dc:map(true):seek(stile.x,stile.y):char(...):map(false)
```

Comment:
Good job with this! Did you test it with TWBT, or just steal it from scripts like gui/siege-engine? I assume it works with TWBT since this is here, but it would be nice to know.

Reply:
I "adapted" (stole) this from a script called guide-path

Reply:
I did not test it with TWBT.

Reply:
Yeah, that's one of the other scripts that got changed along with gui/siege-engine when we added TWBT support to the drawing API (I thought of gui/siege-engine because it was the one that inspired the change). As long as you use this every time you draw to the map area, it should work fine.